src/dsmr/parser.h

```diff
@@ -83,7 +87,11 @@ struct ParsedData<> {
 // Do not use F() for multiply-used strings (including strings used from
 // multiple template instantiations), that would result in multiple
 // instances of the string in the binary
+#ifndef ARDUINO_ARCH_ESP8266
 static constexpr char DUPLICATE_FIELD[] PROGMEM = "Duplicate field";
+#else
+static constexpr char DUPLICATE_FIELD[] = "Duplicate field";
```

Comment:
It may be cleaner, but it won't work as PROGMEM _is_ defined and the
reader.ino example sketch seems to trigger a bug in ESP8266 for Arduino:

Arduino: 1.6.8 (Linux), Board: "NodeMCU 1.0 (ESP-12E Module), 80 MHz,
115200, 4M (3M SPIFFS)"

In file included from
/home/ewasscher/Arduino/libraries/arduino-dsmr/src/dsmr.h:35:0,
                 from
/home/ewasscher/Arduino/arduino-dsmr-read-test/arduino-dsmr-read-test.ino:22:
/home/ewasscher/Arduino/libraries/arduino-dsmr/src/dsmr/parser.h:160:23:
error: dsmr::INVALID_UNIT causes a section type conflict with __c
 static constexpr char INVALID_UNIT[] PROGMEM = "Invalid unit";
                       ^
In file included from
/home/ewasscher/.arduino15/packages/esp8266/hardware/esp8266/2.2.0/cores/esp8266/Arduino.h:244:0,
                 from sketch/arduino-dsmr-read-test.ino.cpp:1:
/home/ewasscher/.arduino15/packages/esp8266/hardware/esp8266/2.2.0/cores/esp8266/pgmspace.h:21:51:
note: '__c' was declared here
 #define PSTR(s) (__extension__({static const char __c[] PROGMEM = (s);
&__c[0];}))
                                                   ^
/home/ewasscher/.arduino15/packages/esp8266/hardware/esp8266/2.2.0/cores/esp8266/WString.h:38:76:
note: in definition of macro 'FPSTR'
 #define FPSTR(pstr_pointer) (reinterpret_cast<const __FlashStringHelper
*>(pstr_pointer))

^
/home/ewasscher/.arduino15/packages/esp8266/hardware/esp8266/2.2.0/cores/esp8266/WString.h:39:34:
note: in expansion of macro 'PSTR'
 #define F(string_literal) (FPSTR(PSTR(string_literal)))
                                  ^
/home/ewasscher/Arduino/arduino-dsmr-read-test/arduino-dsmr-read-test.ino:110:20:
note: in expansion of macro 'F'
       Serial.print(F(": "));
                    ^
exit status 1
Error compiling for board NodeMCU 1.0 (ESP-12E Module).



2016-05-29 12:00 GMT+02:00 Matthijs Kooijman <notifications@github.com>:

> In src/dsmr/parser.h
> <https://github.com/matthijskooijman/arduino-dsmr/pull/3#discussion_r65001027>
> :
>
> >  static constexpr char DUPLICATE_FIELD[] PROGMEM = "Duplicate field";
> > +#else
> > +static constexpr char DUPLICATE_FIELD[] = "Duplicate field";
>
> I guess it might be cleaner to fix this using something like:
>
> #if !defined(PROGMEM)
> #define PROGMEM
> #endif
>
> in an appropriate place (included by everyone that needs it).
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/matthijskooijman/arduino-dsmr/pull/3/files/27e2ace07f866f0285c58f57b6690c9dfc280bab#r65001027>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AB7KB_z7jG3an0dBzbfcYswaeO_INK4qks5qGWPWgaJpZM4Im8LH>
> .
>